R/fetch_county_geo.R

```diff
@@ -86,56 +86,42 @@ fetch_ctu_geo <- function(core = TRUE, ...) {
 
   county_list <- if (core == TRUE) {
     c(
-      "Anoka",
-      "Carver",
-      "Dakota",
-      "Hennepin",
-      "Ramsey",
-      "Scott",
-      "Washington"
+      "003", # "Anoka",
+      "019", # "Carver",
+      "037", # "Dakota",
+      "053", # "Hennepin",
+      "123", # "Ramsey",
+      "139", # "Scott",
+      "163" # "Washington"
     )
   } else if (core == FALSE) {
     c(
-      "Anoka",
-      "Carver",
-      "Dakota",
-      "Hennepin",
-      "Ramsey",
-      "Scott",
-      "Sherburne",
-      "Washington",
-      "Wright"
+      "003", # "Anoka",
+      "019", # "Carver",
+      "037", # "Dakota",
+      "053", # "Hennepin",
+      "123", # "Ramsey",
+      "139", # "Scott",
+      "163", # "Washington"
+      "141", # "Sherburne",
+      "171" # "Wright"
     )
   }
 
   cities <- tigris::county_subdivisions(
-    state = "MN",
+    state = 27,
     county = county_list,
-    class = "sf"
+    class = "sf",
+    ...
   ) %>%
-    mutate(NAME = case_when(
-      LSAD == 44 ~ paste(NAME, "Twp."),
-      LSAD == 46 ~ paste(NAME, "(unorg.)"),
-      TRUE ~ NAME
-    )) %>%
-    ## if expanding to greater mn or another region, you do have to do some unions, and further cleaning.
-    #   group_by(NAME) %>%
-    #   mutate(n = n()) %>%
-    #   left_join(st_drop_geometry(county_outline) %>%
-    #               transmute(
-    #                 COUNTYFP = COUNTYFP,
-    #                 CONAME = NAME
-    #               )) %>%
-    #   mutate(NAME = case_when(
-    #     n > 1 & LSAD != 25 ~ paste0(NAME, " - ", CONAME, " Co."), # cities dont get merged
-    #     TRUE ~ NAME
-    #   )) %>%
-    #   group_by(NAME) %>%
-    #   summarise() %>%
-    #   # summarize(geometry = st_union(geom)) %>%
-    #   arrange(NAME) %>%
-    #   rename(GEO_NAME = NAME)
-    transmute(
+    dplyr::mutate(
+      NAME = dplyr::case_when(
+        LSAD == 44 ~ paste(NAME, "Twp."),
+        LSAD == 46 ~ paste(NAME, "(unorg.)"),
+        TRUE ~ NAME
+      )
+    ) %>%
+    dplyr::transmute(
```

Comment:
@eroten I made some changes here. I was sloppy before, sorry! If core == FALSE, some merging of cities across boundaries needs to happen. Hanover is a good example of this. It should be fixed now! When core == FALSE, the function is slower, but now returns better geographies :)

Reply:
just fyi